tls/s2n_record_read.c

```diff
@@ -72,6 +72,15 @@ int s2n_record_header_parse(struct s2n_connection *conn, uint8_t * content_type,
 
     uint8_t version = (protocol_version[0] * 10) + protocol_version[1];
 
+    uint8_t tls_major_version = protocol_version[0];
+    /* TLS servers compliant with this specification MUST accept any value {03,XX} as the record layer version number
+     * for ClientHello.
```

Comment:
I think so, we are still being liberal in what we accept. 

Our previous behavior was to completely ignore the Record Layer Protocol Version for the Client Hello, and our new behavior is to validate that the TLS Major Version at least matches any of the known valid major versions used in TLS's history.